src/main/java/menion/android/whereyougo/gui/activity/XmlSettingsActivity.java

```diff
@@ -158,79 +158,49 @@ public boolean onPreferenceClick(Preference preference) {
         boolean status = false;
         String key = preference.getKey();
 
-        if (key.equals("")) {
-            // DO NOTHING
-        } else if (key.equals(getString(R.string.pref_KEY_S_ROOT))) {
+        if (key.equals(getString(R.string.pref_KEY_S_ROOT))) {
             UtilsGUI.dialogDoItem(this, getText(R.string.pref_root), R.drawable.var_empty, getText(R.string.pref_root_desc),
                     getString(R.string.cancel), null,
-                    getString(R.string.folder_select), new DialogInterface.OnClickListener() {
-                        @Override
-                        public void onClick(DialogInterface dialog, int which) {
-                            FileChooserDialog selectDialog = new FileChooserDialog(XmlSettingsActivity.this);
-                            selectDialog.loadFolder(Preferences.GLOBAL_ROOT);
-                            selectDialog.setFolderMode(true);
-                            selectDialog.setCanCreateFiles(false);
-                            selectDialog.setShowCancelButton(true);
-                            selectDialog.addListener(new FileChooserDialog.OnFileSelectedListener() {
-                                public void onFileSelected(Dialog source, File folder) {
-                                    source.dismiss();
-                                    if (((MainApplication) A.getApp()).setRoot(folder.getAbsolutePath())) {
-                                        PreviewPreference preferenceRoot = (PreviewPreference) findPreference(R.string.pref_KEY_S_ROOT);
-                                        preferenceRoot.setValue(FileSystem.ROOT);
-                                        MainActivity.refreshCartridges();
-                                    }
+                    getString(R.string.folder_select), (dialog, which) -> {
+                        FileChooserDialog selectDialog = new FileChooserDialog(XmlSettingsActivity.this);
+                        selectDialog.loadFolder(Preferences.GLOBAL_ROOT);
+                        selectDialog.setFolderMode(true);
+                        selectDialog.setCanCreateFiles(false);
+                        selectDialog.setShowCancelButton(true);
+                        selectDialog.addListener(new FileChooserDialog.OnFileSelectedListener() {
+                            public void onFileSelected(Dialog source, File folder) {
```

Comment:
Also move to lambda.

Reply:
At least to my knowledge this won't work because of the second method we are declaring below this.

Reply:
Then let it as is. No must do here.